### ISSUE TITLE
Allowlist/block hosts directly from log page

### ISSUE THREAD

**Issue by w4tadrag** [celzero/rethink-app#208]
There are times when I need to whitelist & block certain links/hosts, but I can't find the option for it. I would love it if we can block/whitelist directly from log page. Also have sections in the settings for whitelist and block list to manually add links to block or whitelist. These sections should have a list of what I have blocked or whitelisted and I if want to disable these links I can do it from these sections. Hopefully this makes sense. For your reference, Blokada has something like this.

**Comment by ignoramous** [comment]
This is a top ask from users and we are working on-priority on this.

It is a bit of a rewrite so, this feature is a month away at best.

**Comment by HyknuF** [comment]
First of all, thank you guys so much for this amazing app and your lovelly work. i really love it!
I can't add any decent to this issue, but to really looking forward to this feature. Thanks in advance!

**Comment by ignoramous** [comment]
> It is a bit of a rewrite so, this feature is a month away at best.

The rewrite has begun. The feature _may be_ a month-away at best (:

See also: #18

**Comment by ignoramous** [comment]
> The rewrite has begun. The feature may be a month-away at best (:


It has been a *year*! But: It is a month away at best ;) Only UI changes pending for Husain to impl, as network engine changes are done (though, untested and rough).

**Comment by jerryn70** [comment]
@ignoramous option to add whitelist from url and from file (.txt) similar like adaway and dns66 is also a good option. In my point of view it's a must have feature.

**Comment by hussainmohd-a** [comment]
One more issue is open to import domains and IPs from url or file. Closing this issue as it has been resolved. 

One more issue was opened to import domains and IPs from a URL or file [#794](https://github.com/celzero/rethink-app/issues/794). We have implemented a solution to address this and are closing this issue as resolved. Ref a64f10326bdea1dd7d3a8247d1135a177c2c1610, db52551ceb40da0430a2fa356647132cf9938263, ed4e2f81baddab57d792bcf185537fe842d46457, 83b57fbd01865d4be9a68942ef5260cee88e8357
